app/controllers/metrics_controller.rb

```diff
@@ -0,0 +1,359 @@
+      end
+    end
+    @parsed_data.each do |author, commits|
+      @parsed_data[author] = Hash[commits.sort_by {|date, _commit_count| date }]
+      @total_commits += commits.inject (0) {|sum,value| sum + value[1] }
+    end
+  end
+
+  ######################## HTTP Query Execution #########################
+
+  # make the actual github api request with graphql and query message.
+  # data: the query message made in get_query method. Documented in detail in get_query method
+  def query_commit_statistics(data)
+    uri = URI.parse("https://api.github.com/graphql")
+    http = Net::HTTP.new(uri.host, uri.port) # host: api.github.com, port: 443
+    http.use_ssl = true
+    http.verify_mode = OpenSSL::SSL::VERIFY_PEER
+    request = Net::HTTP::Post.new(uri.path, 'Authorization' => 'Bearer' + ' ' + session["github_access_token"]) # set up authorization
+    request.body = data.to_json # convert query message to json and pass as request body
+    #    http.request(request)
+    response = http.request(request) # make the actual request
+    ActiveSupport::JSON.decode(response.body.to_s) # convert the response body to string, decoded then return
+  end
+
+  # pr_object contains head commit reference num, author name, and repo name
+  # using the github api end point to get the pull request status info
+  def query_pull_request_status(pr_object)
+    url = "https://api.github.com/repos/" + pr_object[:owner] + "/" + pr_object[:repository] + "/commits/" + pr_object[:head_commit] + "/status"
+    ActiveSupport::JSON.decode(Net::HTTP.get(URI(url)))
+  end
+
+  # Handle the create action for a github metric, which stores a datapoint mapping a team id, and a github email address
+  # to an expertiza User, with a datapoint for their total contributions to the project. Users are asked to create the
+  # mapping from their Github email within their user profile, but we also try to intelligently determine that mapping if
+  # the user has not provided an email, and their profile contains enough clues.
+  def create_github_metric(team_id, github_id, total_commits)
+    metric = Metric.where("team_id = ? AND github_id = ?", team_id, github_id).first
+    # Attempt to find user by their github email -- Mapping already exists
+    user = User.find_by_github_id(github_id)
+
+    # If mapping does not exist, attempt to figure out their github email from the information we have
+    if user.nil?
+      email = github_id.split('@')
+      #Check if NCSU email
+      if email[1] == 'ncsu.edu'
+        user = User.find_by_email(github_id)
+        # If success, go ahead and save this mapping for future queries
+        user.github_id = github_id unless user.nil?
+        user.save unless user.nil?
+      else # Try mapping from unityID@gmail.com or unityID@anotherprovider.com
+        user = User.find_by_email(email[0] + "@ncsu.edu")
+        # If success, go ahead and save this mapping for future queries
+        user.github_id = github_id unless user.nil?
```

Comment:
are there some common providers we could check for? 
perhaps gmail and icloud should cover most i would think